embassy-rp/src/spi.rs

```diff
@@ -18,6 +18,49 @@ pub enum Error {
     // No errors for now
 }
 
+#[derive(Clone, Copy, PartialEq, Eq)]
+// #[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct Builder {
+    pub frequency: Option<u32>,
+    pub phase: Option<Phase>,
+    pub polarity: Option<Polarity>,
+}
+
+impl Builder {
+    pub fn frequency(self, frequency: u32) -> Self {
```

Comment:
I would personally think having these setter functions directly on the config struct is nicer.. such that you do ‘Config::default().frequency(1234)‘ and don't need the ‘build()‘ nor the builder struct